### PR TITLE
[coqnative] Exit with non zero code on error

### LINKED ISSUE #14472: coqnative exits with code 0 on error
This doesn't stop make compilation for instance and the error can then easily go unnoticed.

### PR DESCRIPTION
**Pull request by proux01**
**Kind:** bug fix

Fixes #14472

### CODE REVIEW

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Test-suite not happy, for a seemingly unrelated issue?

**Comment by proux01** [comment]
No, this is a bug/difference in behavior between coqc and coqnative that went unnoticed because of #14472
The following used to succeed
```
coqc  -q -I src  -R theories '' -R src '' theories/test.v
```
whereas
```
coqnative -I src  -R theories '' -R src '' theories/test.v
```
fails with `Error: The empty string is not an identifier.`.

**Comment by proux01** [comment]
Should be fixed.
BTW @ppedrot why is there a reimplementation of dirpath_of_string that already exists in libnames? (your implementation looks much nicer though)

**Comment by ppedrot** [comment]
I think that coqnative was written to only depend on the kernel, like coqchk, and library is way above that.

**Comment by SkySkimmer** [comment]
>(your implementation looks much nicer though)

`String.split_on_char` used in the coqnative implementation is a OCaml 4.04 (2016) addition.
The libnames implementation has been there since 12965209478bd99dfbe57f07d5b525e51b903f22 (2002).

**Comment by proux01** [comment]
Thanks

**Comment by ejgallego** [comment]
> BTW @ppedrot why is there a reimplementation of dirpath_of_string that already exists in libnames? (your implementation looks much nicer though)

We hope to consolidate these functions into a `coq.loadpath` lib that lives in the right level, so that should eventually be solved.

File: topbin/coqnative_bin.ml [hunk 1]
```diff
@@ -322,4 +322,6 @@ let () =
     let senv = init_coq () in
     compile senv ~in_file
   with exn ->
-    Format.eprintf "Error: @[%a@]@\n%!" Pp.pp_with (CErrors.print exn)
+    Format.eprintf "Error: @[%a@]@\n%!" Pp.pp_with (CErrors.print exn);
+    let exit_code = if (CErrors.is_anomaly exn) then 129 else 1 in
```
Comment (silene) [inline topbin/coqnative_bin.ml:326]: Has this been discussed somewhere? The convention on Unix is that exit code 128+N means that the signal N caused an abrupt termination of the process. So, using 129 might confuse shells, scripts, etc.

Comment (ppedrot) [reply topbin/coqnative_bin.ml:326]: This is the convention we already use for other coq binaries.

Comment (proux01) [reply topbin/coqnative_bin.ml:326]: This seems to date back from a while ago https://github.com/coq/coq/commit/78793236fa289d6e48cb10c9694ee493e0952c28

**Pull request merged**